doc/source/whatsnew/v0.21.0.txt

```diff
@@ -79,6 +79,62 @@ Other Enhancements
 - :func:`date_range` now accepts 'YS' in addition to 'AS' as an alias for start of year (:issue:`9313`)
 - :func:`date_range` now accepts 'Y' in addition to 'A' as an alias for end of year (:issue:`9313`)
 
+.. _whatsnew_0210.enhancements.read_csv:
+
+``read_csv`` use `python-requests` (if installed) to support basic auth and much more
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If `python-requests` library is installed try to use it first. If not, continue using urllib
```

Comment:
Lets say I am a heavy duty pandas user with http (dont use basic auth. Just use plain old http). I get it that possibly in future requests might become the default (if installed). I want to be prepared and find issues early. How should i force pandas to use requests, even if I am not using http auth? How else do you expect to gain confidence in requests and detect scenarios of inconsistencies?
> The condition to use requests should be if url_params is a non-empty dictionary

how about: `url_params = { 'use_requests' : True }` forces use of requests even if it is not `auth` or `verify` scenario.